src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -92,7 +91,54 @@ def _get_symbol_table(node):
     return None
 
 
-def _check_bound_is_full_extent(array, dim, index, operator):
+def _check_args(array, dim):
+    '''Utility routine used by the _check_bound_is_full_extent and
+    _check_array_range_literal functions to check common arguments.
+
+    This routine is only in fparser2.py until #717 is complete as it
+    is used to check that array syntax in a where statement is for the
+    full extent of the dimension. Once #717 is complete this routine
+    can be removed.
+
+    :param array: the node to check.
+    :type array: :py:class:`pysclone.psyir.node.array`
+    :param int dim: the dimension index to use.
+
+    :raises TypeError: if the supplied arguments are of the wrong type.
+    :raises ValueError: if the value of the supplied dim argument is \
+        less than 1 or greater than the number of dimensions in the \
+        supplied array argument.
+
+    '''
+    if not isinstance(array, Array):
+        raise TypeError(
+            "method _check_args 'array' argument should be an "
+            "Array type but found '{0}'.".format(type(array).__name__))
+
+    if not isinstance(dim, int):
+        raise TypeError(
+            "method _check_args 'dim' argument should be an "
+            "int type but found '{0}'.".format(type(dim).__name__))
+    if dim < 1:
+        raise ValueError(
+            "method _check_args 'dim' argument should be at "
+            "least 1 but found '{0}'.".format(dim))
+    if dim > len(array.children):
+        raise ValueError(
+            "method _check_args 'dim' argument should be at "
+            "most the number of dimensions of the array '{0}' but found "
```

Comment:
I don't think we should quote integer values. Maybe use parentheses around {0} but leave {1} 'bare'.

Reply:
Updated.

Reply:
Almost. Please change "'{0}'" to "({0})".

Reply:
Done.